lib/phoenix/test/channel_test.ex

```diff
@@ -157,6 +157,19 @@ defmodule Phoenix.ChannelTest do
   end
 
   @doc """
+  Same as subscribe_and_join/3 but returns either the socket or throws an error
+
+  This is helpful when you are not testing authentication and just need the
+  socket.
+  """
+  defmacro subscribe_and_join!(channel, topic, payload \\ Macro.escape(%{})) do
+    quote do
+      {:ok, _, socket} = subscribe_and_join(@endpoint, unquote(channel), unquote(topic), unquote(payload))
```

Comment:
We need to use a `case` and raise a nice error message. something like: raise "could not join channel, got error: #{inspect error}" would be enough.